addons/l10n_vn/migrations/17.0.2.0.2/post-migration.py

```diff
@@ -0,0 +1,27 @@
+# -*- coding: utf-8 -*-
+from odoo import api, SUPERUSER_ID
+from odoo.osv import expression
+
+FIXED_ACCOUNTS_TYPE = {
+    'asset_prepayments': ['242'],
+    'expense_depreciation': ['6274', '6414', '6424'],
+}
+
+
+def _fix_accounts_type(env):
+    for company in env['res.company'].with_context(active_test=False).search([('chart_template', '=', 'vn')]):
+        for correct_account_type, accounts_code in FIXED_ACCOUNTS_TYPE.items():
+            doamin = expression.AND([
+                [('company_id', '=', company.id), ('account_type', '!=', correct_account_type)],
+                expression.OR([
+                    [('code', 'like', code + '%')] for code in accounts_code
+                ])
+            ])
+            accounts = env['account.account'].search(doamin)
+            if accounts:
+                accounts.write({'account_type': correct_account_type})
```

Comment:
summary of the suggested changes:
- correcting typo `doamin` -> `domain`
- concatenation to f-string
- query all company accounts at the same time to make it faster in case there are MANY companies
- no need to `write`, `record.property = stuff` does the trick
- no need to `if record: record.property = stuff`, if the recordSet is empty, the ORM doesn't write anything

```suggestion
def _fix_accounts_type(env):
    for correct_account_type, accounts_code in FIXED_ACCOUNTS_TYPE.items():
        domains_per_company = []
        for company in env['res.company'].with_context(active_test=False).search([('chart_template', '=', 'vn')]):
            domain = expression.AND([
                [('company_id', '=', company.id), ('account_type', '!=', correct_account_type)],
                expression.OR([
                    [('code', 'like', f'{code}%')] for code in accounts_code
                ]),
            ])
            domains_per_company.append(domain)
        accounts = env['account.account'].search(expression.OR(domains_per_company))
        accounts.account_type = correct_account_type
```